src/client/threads.js

```diff
@@ -318,13 +319,47 @@ NetsProcess.prototype.parseRPCResult = function (result) {
     return result;
 };
 
+function listToJSON(list) {
```

Comment:
Maybe we can move it into a `utils.js` so that it's not global anymore and easy to reuse